types/trigger.d.ts

```diff
@@ -40,7 +40,7 @@ export type OutputStrings = { [outputKey: string]: LocaleText | string };
 export type Output = {
   responseOutputStrings: OutputStrings;
 } & {
-  [key: string]: (params?: { [param: string]: string | number | undefined }) => string;
+  [key: string]: (params?: { [param: string]: string | number | object | undefined }) => string;
```

Comment:
Should `object` be more specific here, e.g. `{ [key: string]: string }` (which apparently does still let you assign a function to `toString`, although we could be explicit about that too).

Reply:
This typing (and some other added/adjusted typing) are preliminary, My end goal is to have something like `type OutputStringsObject = <some object definition that requires a `toString` implementation, and whose properties must be string or number>`. I opened this PR early in the dev process to make sure that we were on the same page as far as a direction to take this before I invested too much time/effort into the process.

As for `foo.bar.baz`, that would be solved by a recursive function as I mentioned in the PR comment, but I was unsure if that was a desired behavior or not, and it's a bit more effort, so I didn't implement it yet.